platform/src/server/http/router/router.ts

```diff
@@ -128,34 +139,30 @@ export class Router {
       kibanaRequest = KibanaRequest.from(request, routeSchemas);
     } catch (e) {
       // TODO Handle failed validation
-
-      response.status(400);
-      response.json({ error: e.message });
-      return;
+      return responseToolkit.response({ error: e.message }).code(400);
     }
 
     try {
       const kibanaResponse = await handler(kibanaRequest, responseFactory);
 
       if (kibanaResponse instanceof KibanaResponse) {
-        response.status(kibanaResponse.status);
-
-        if (kibanaResponse.payload === undefined) {
-          response.send();
-        } else if (kibanaResponse.payload instanceof Error) {
+        let payload = null;
+        if (kibanaResponse.payload instanceof Error) {
           // TODO Design an error format
-          response.json({ error: kibanaResponse.payload.message });
-        } else {
-          response.json(kibanaResponse.payload);
+          payload = { error: kibanaResponse.payload.message };
```

Comment:
Looks like we're treating an `instanceof Error` automatically as a `Boom` instance, but that's not guaranteed right?

Reply:
Do you think that something like [`Boom.boomify()`](https://github.com/hapijs/boom#boomifyerr-options) is appropriate here and in the error handler below?

Reply:
> Looks like we're treating an `instanceof Error` automatically as a `Boom` instance, but that's not guaranteed right?

Hmm, nope, there is no assumption about Boom here, what do you mean?

> Do you think that something like Boom.boomify() is appropriate here and in the error handler below?

Well, we haven't used `Boom` in the new platform yet and relied on very naive error handling approach until now (manually set error code and `{error: ....}` body). Do you think we should bring that dependency in? I didn't work with Boom too much, to be sure that we really need it (at least in this PR).